package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -1002,32 +1002,53 @@ export const MessageInputProvider = <
   }) => {
     const id = generateRandomId();
     const mimeType = lookup(file.name);
+
+    const blockedFile = blockedFiles?.some((fileExtensionType: string) =>
+      file.name?.includes(fileExtensionType),
+    );
+    const blockedMime = blockedMimeFiles?.some((mimeType: string) => file.name?.includes(mimeType));
+
     const newFile = {
       file: { ...file, type: mimeType || file?.type },
       id,
-      state: FileState.UPLOADING,
+      state: blockedFile || blockedMime ? FileState.NOT_SUPPORTED : FileState.UPLOADING,
```

Comment:
Please extract this ternary out to a variable

Reply:
✅